### PR TITLE
J2DPane.cpp

### PR DESCRIPTION
**Pull request by notyourav**
was gross

include/JSystem/J2DGraph/J2DPane/J2DPane.h
include/JSystem/geometry.h
libs/JSystem/J2DGraph/J2DPane.cpp

### CODE REVIEW

File: include/JSystem/J2DGraph/J2DPane/J2DPane.h [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+extern f32 lbl_80456208;  // 0.5f
+extern u8 lbl_8045620C;
+extern u8 lbl_80456210;
+extern u8 lbl_80456214;
+extern u8 lbl_80456218;
+
+using namespace JGeometry;
+extern TBox2<f32> lbl_804349B0;
+
+typedef enum J2DRotateAxis {
```
Comment (lioncash) [inline include/JSystem/J2DGraph/J2DPane/J2DPane.h:90]: Small difference between C and C++. `typedef` on enums and structs isn't needed in C++.

File: include/JSystem/J2DGraph/J2DPane/J2DPane.h [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    BOTTOM_RIGHT = 8,
+};
+
+struct JUTResReference {
+    JUTResReference() : res(NULL) {}
+    void* getResource(JSURandomInputStream* stream, u32 size, JKRArchive* archive);
+    void* res;
+};
+
+struct J2DGrafContext {};
```
Comment (lioncash) [inline include/JSystem/J2DGraph/J2DPane/J2DPane.h:114]: You may want to consider putting this and all the tiny structs below into their own headers, J2DScreen for example is quite large, so placing it here isn't particularly advisable.

This'd make for way less code shifting in the future (minimizing conflict potential), and makes reviewing a little nicer for code in the future.

e.g. All the J2DAnm stuff can go in its own header, given it's used extensively in other pieces of code, not just the J2DPane class hierarchy.

File: include/JSystem/J2DGraph/J2DPane/J2DPane.h [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+struct J2DScreen {
```
Comment (lioncash) [inline include/JSystem/J2DGraph/J2DPane/J2DPane.h:116]: This should also be a class

File: include/JSystem/J2DGraph/J2DPane/J2DPane.h [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+    static u8* getDataManage() { return lbl_80451590; }
+};
+
+struct J3DTransformInfo {};
```
Comment (lioncash) [inline include/JSystem/J2DGraph/J2DPane/J2DPane.h:120]: This and many of the structs below should actually be classes

File: include/JSystem/J2DGraph/J2DPane/J2DPane.h [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+    /* 0x004 */ u16 _4;
+    /* 0x006 */ u8 _6[2];
+    /* 0x008 */ int id;
+    /* 0x00c */ u8 _c[4];
+    /* 0x010 */ u64 _10;
+    /* 0x018 */ u64 _18;
+    /* 0x020 */ TBox2<f32> _20;
+    /* 0x030 */ TBox2<f32> _30;
+    /* 0x040 */ TBox2<f32> _40;
+    /* 0x050 */ u8 _50[96];
```
Comment (lioncash) [inline include/JSystem/J2DGraph/J2DPane/J2DPane.h:226]: These are two [Mtx](https://github.com/zeldaret/tp/blob/master/include/mtx_vec.h#L14) instances.

File: include/JSystem/J2DGraph/J2DPane/J2DPane.h [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+    /* 0x0b0 */ u8 _b0;
```
Comment (lioncash) [inline include/JSystem/J2DGraph/J2DPane/J2DPane.h:227]: This is a `bool` indicating visibility.

File: include/JSystem/J2DGraph/J2DPane/J2DPane.h [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+    /* 0x0b1 */ u8 mCullMode;
+    /* 0x0b2 */ u8 _b2;
```
Comment (lioncash) [inline include/JSystem/J2DGraph/J2DPane/J2DPane.h:229]: This `u8` represents alpha

File: include/JSystem/J2DGraph/J2DPane/J2DPane.h [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+    /* 0x0c4 */ f32 _c4;
+    /* 0x0c8 */ f32 _c8;
+    /* 0x0cc */ f32 _cc;
+    /* 0x0d0 */ f32 _d0;
+    /* 0x0d4 */ f32 _d4;
+    /* 0x0d8 */ f32 _d8;
+    /* 0x0dc */ JSUTree<J2DPane> mPaneTree;
+    /* 0x0f8 */ const J2DAnmTransform* mTransform;
+    /* 0x0fc */ u8 _fc[0x104];
+    /* 0x200 */ f32 _200;
```
Comment (lioncash) [inline include/JSystem/J2DGraph/J2DPane/J2DPane.h:247]: Is a J2DPane truly 0x200 bytes in size? iirc it was only 0x100 (based off the allocation near the bottom of `dFile_select_c::screenSet`

File: include/JSystem/J2DGraph/J2DPane/J2DPane.h [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+    /* 0x14 */ virtual void add(f32 x, f32 y);
+    /* 0x18 */ virtual void resize(f32 x, f32 y);
+    /* 0x1c */ virtual void setCullBack(bool cull);
+    /* 0x20 */ virtual void setCullBack(GXCullMode cmode);
+    /* 0x24 */ virtual void setAlpha();
+    /* 0x28 */ virtual bool setConnectParent(bool connected);
+    /* 0x2c */ virtual void calcMtx();
+    /* 0x30 */ virtual void update();
+    /* 0x34 */ virtual void drawSelf(f32 arg1, f32 arg2);
+    /* 0x38 */ virtual void drawSelf(f32 arg1, f32 arg2, f32 arg3[2][3][4]);
```
Comment (lioncash) [inline include/JSystem/J2DGraph/J2DPane/J2DPane.h:161]: `arg3` here is likely a pointer to a [Mtx](https://github.com/zeldaret/tp/blob/master/include/mtx_vec.h#L14)

**Review by Pheenoh — APPROVED** [review]

**Pull request merged**